internal/operator/controller/contour/service.go

```diff
@@ -43,6 +43,62 @@ const (
 	// TODO [danehans]: Make proxy protocol configurable or automatically enabled. See
 	// https://github.com/projectcontour/contour-operator/issues/49 for details.
 	awsLbBackendProtoAnnotation = "service.beta.kubernetes.io/aws-load-balancer-backend-protocol"
+	// envoyServiceHTTPPort is the HTTP port number of the Envoy service.
+	envoyServiceHTTPPort = 80
+	// envoyServiceHTTPSPort is the HTTPS port number of the Envoy service.
+	envoyServiceHTTPSPort = 443
+	// envoyNodePortHTTPPort is the NodePort port number for Envoy's HTTP service. For NodePort
+	// details see: https://kubernetes.io/docs/concepts/services-networking/service/#nodeport
+	envoyNodePortHTTPPort = 30080
```

Comment:
Why set these statically? There's a possibility of conflict with another nodePort type service.

Reply:
@stevesloka I'm trying to avoid exposing nodeport configuration during the initial release of the Network Publishing API. The API changes were already quite large without exposing nodeport configuration. I would like to revisit the topic when users identify nodeport configuration as an issue or a desired capability.

Reply:
Sure that's fine, but is it better to not define NodePorts and let Kubernetes define which it should use?

Reply:
@stevesloka thank you for the feedback.  I originally thought it would be beneficial to have a well-known nodeport, but I created https://github.com/projectcontour/contour-operator/issues/184 to surface this information in status.  I have removed the static nodeport assignment for the Envoy Service, but kept the port assignments for the kind cluster script.

Reply:
@stevesloka after further investigation, we need the ability to assign the nodeport through the Network Publishing API or statically set a port. This is because the kind config requires port mapping definitions:
```
  extraPortMappings:
  - containerPort: 30080
    hostPort: 80
    listenAddress: "0.0.0.0"
  - containerPort: 30443
    hostPort: 443
    listenAddress: "0.0.0.0"
```

Without these ports being used by a `Contour`, the e2e tests fail. I'm going to revert the latest commit so static nodeports are used and expose the nodeport configuration in a subsequent PR.